packages/admin-panel/src/pages/ChangePasswordPage.js

```diff
@@ -1,23 +1,156 @@
-/**
- * Tupaia MediTrak
- * Copyright (c) 2017 Beyond Essential Systems Pty Ltd
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
-import React from 'react';
+import React, { useState, lazy, Suspense } from 'react';
 import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import MuiDivider from '@material-ui/core/Divider';
+import { connect } from 'react-redux';
+import { useForm } from 'react-hook-form';
+import { Button, TextField, SmallAlert, PasswordStrengthBarFallback } from '@tupaia/ui-components';
 import { usePortalWithCallback } from '../utilities';
 import { Header } from '../widgets';
+import { updatePassword, getUser } from '../authentication';
+
+// Lazy load the component as it uses zxcvbn which is a large library.
+// For more about lazy loading components @see: https://reactjs.org/docs/code-splitting.html#reactlazy
+const PasswordStrengthBar = lazy(() => import('../widgets/PasswordStrengthBar'));
```

Comment:
Really cool! Nice to see that lazy loading is easy in react :-)

Reply:
I'm interested to know more about the code splitting. I imagine:
- because we just build a single bundle for ui-components, the javascript that represents our customised PasswordStrengthBar component has already been downloaded at this point; but
- it hasn't loaded in the dependencies of that until that part of our ui-components bundle is actually run, so the zxcvbn gets lazily loaded

Is that right?

Would be nice to code split the ui-components bundle itself at some point!

Reply:
thanks for the question. It made me realise that the lazy loading wasn't working properly and triggered me to fix it. As we discussed the best solution for this use case was to bring the PasswordStrengthBar into the admin-panel package so that it could be properly lazy loaded. Yes it could be good to split the ui-components at some point. I have created a task in the backlog for that and added some notes from my investigation. @see https://github.com/beyondessential/tupaia-backlog/issues/1347